Data-Structures/src/main/java/linkedlist/LinkedList.java

```diff
@@ -40,21 +40,19 @@ public void append(T value){
     while(this.current.next != null){
       this.current = current.next;
     }
-    Node <T> newNode = new Node(value, null);
-    current.next = newNode;
+    current.next = new Node(value, null);
   }
 
   public void insertBefore(T value, T target){
     if(head == null){
       head = new Node(value, null);
     }
     this.current = head;
-    while(current.next != null){
-      if(current.value == target){
-        Node newNode = new Node(value, current);
-        current.next = newNode;
+    while(current != null) {
+      if(current.next.value == target){
+        current.next = new Node(value, this.current.next);
       }
-      current = current.next;
+      break;
```

Comment:
Because this `break` statement is here this loop will only ever run once. This means that this method only works if the user is trying to insert a value before the first or second value in the linked list.